hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java

```diff
@@ -649,6 +651,7 @@ private SqlPlanImpl toPlan(
             );
         } else if (physicalRel instanceof SinkMapPhysicalRel) {
             assert !isCreateJob;
+            checkIMapByKeyPlanIsAnalyzed(analyze);
             SinkMapPhysicalRel sink = (SinkMapPhysicalRel) physicalRel;
```

Comment:
which test checks this part?

Reply:
https://github.com/hazelcast/hazelcast/pull/26002/commits/a3b54c6c0f17406a240288c3edbcb514509f17fa